frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -912,42 +932,213 @@ function ReaderFooter:addToMainMenu(menu_items)
+                    },
+                    {
+                        text = _("Alongside items"),
+                        checked_func = function()
+                            return not self.settings.progress_bar_position
+                        end,
+                        callback = function()
+                            self.settings.progress_bar_position = nil
+                            self:refreshFooter(true, true)
+                        end
+                    },
+                },
             },
             {
-                text = _("Progress bar on separate line"),
-                checked_func = function()
-                    return self.settings.progress_bar_separate_line
+                text_func = function()
+                    if self.settings.progress_style_thin then
+                        return _("Style: thin")
+                    else
+                        return _("Style: thick")
+                    end
                 end,
                 enabled_func = function()
                     return not self.settings.disable_progress_bar
                 end,
-                callback = function()
-                    self.settings.progress_bar_separate_line = not self.settings.progress_bar_separate_line
-                    self:refreshFooter()
-                    self.ui:handleEvent(Event:new("SetPageBottomMargin", self.view.document.configurable.b_page_margin))
-                    UIManager:setDirty(nil, "ui")
+                sub_item_table = {
+                    {
+                        text = _("Thick"),
+                        checked_func = function()
+                            return not self.settings.progress_style_thin
+                        end,
+                        callback = function()
+                            self.settings.progress_style_thin = nil
+                            local bar_height = self.settings.progress_style_thick_height or PROGRESS_BAR_STYLE_THICK_DEFAULT_HEIGHT
+                            self.progress_bar:updateStyle(true, bar_height)
+                            self:setTocMarkers()
+                            self:refreshFooter(true, true)
+                        end,
+                    },
+                    {
+                        text = _("Thin"),
+                        checked_func = function()
+                            return self.settings.progress_style_thin
+                        end,
+                        callback = function()
+                            self.settings.progress_style_thin = true
+                            local bar_height = self.settings.progress_style_thin_height or PROGRESS_BAR_STYLE_THIN_DEFAULT_HEIGHT
+                            self.progress_bar:updateStyle(false, bar_height)
+                            self:refreshFooter(true, true)
+                        end,
+                    },
+                    {
+                        text = _("Set size"),
+                        separator = true,
```

Comment:
This menu deserves to have added `keep_menu_open = true,` - so you can quickly get back after you press OK to try another value :)
(And may be move `separator=true` at the end of the item, after the `callback=` - as it looks more like how it will appear, below the stuff :) and it's how we do everywhere I think.